icu4c/source/common/unicode/unistr.h

```diff
@@ -3101,7 +3104,8 @@ class U_COMMON_API UnicodeString : public Replaceable
    * must be NUL (U+0000) terminated.
    * @stable ICU 2.0
    */
-  UNISTR_FROM_STRING_EXPLICIT UnicodeString(const char16_t *text);
+  UNISTR_FROM_STRING_EXPLICIT UnicodeString(const char16_t *text) :
```

Comment:
The old from-pointer constructors are not "explicit" by default, and many users probably use the implicit conversion. (Inside ICU we make it explicit.)

The new from-string_view constructors are explicit, so this change would break many users.

Should we use UNISTR_FROM_STRING_EXPLICIT on the new constructor, now that it becomes the successor for the old ones? That means that the default usage would hide the string contents copy when constructing a string_view from something other than a raw pointer. Win some, lose some?

Reply:
The constructor proposed in this PR works by not being marked `explicit`, that's what makes it possible for it to be correctly invoked on existing code like `UnicodeSet(u"[∞]", status)` which wouldn't be possible otherwise.

Reply:
I saw that you were modifying the new templatized constructor but missed that you were removing the "explicit inline" part. I suggest that we put UNISTR_FROM_STRING_EXPLICIT on the replacement then, so that people who did turn that on don't backslide.

Reply:
Done.